packages/block-library/src/navigation/edit/placeholder/index.js

```diff
@@ -100,18 +99,15 @@ export default function NavigationPlaceholder( {
 
 	const [ menuName, setMenuName ] = useState( '' );
 
-	const [ isNewMenuModalVisible, setIsNewMenuModalVisible ] = useState(
-		false
-	);
-
-	const [ createEmpty, setCreateEmpty ] = useState( false );
-
 	const { saveEntityRecord } = useDispatch( coreStore );
 
 	// This callback uses data from the two placeholder steps and only creates
 	// a new navigation menu when the user completes the final step.
 	const createNavigationMenu = useCallback(
-		async ( title = __( 'Untitled Navigation Menu' ), blocks = [] ) => {
+		async ( title, blocks = [] ) => {
+			if ( ! title ) {
+				return;
+			}
```

Comment:
Testing and this will return `undefined` here because there's now no title. It looks like the way this is currently works in the PR is to trigger showing the `UnsavedInnerBlocks` component and have that create the `wp_navigation` post. I really think it'd be best to rely less on `UnsavedInnerBlocks` for that—that component was designed to handle a specific edge case for patterns and not a whole lot else.

I feel like it'd be better to refactor or duplicate the code that generates a title in `UnsavedInnerBlocks` so that it can be used in the Placeholder too. That way there's always a title when we call `createNavigationMenu` and the change is a bit less involved.

It shouldn't be too much to achieve that, just call the `useTemplatePartName` hook in the Placeholder component to get the area name and move this bit of code here into a shared function (or duplicate it) so that it can be used by both `UnsavedInnerBlocks` and `NavigationPlaceholder`:
https://github.com/WordPress/gutenberg/blob/88883a3ea23f5b3d8aff563f6a8447e4133b9e55/packages/block-library/src/navigation/edit/unsaved-inner-blocks.js#L137-L161

Reply:
I added `useCreateNavigationMenu` that can be reused in both places. The process doesn't fully work yet and there is some flickering involved, but it's one step closer now. cc @tellthemachines

Reply:
Thanks @adamziel ! I fixed all the bugs I could find and everything seems to be working correctly now. Would appreciate some extra testing 😄